modules/carto/src/api/maps-v3-client.ts

```diff
@@ -134,13 +136,18 @@ function getParameters({
   columns,
   clientId,
   aggregationExp,
-  aggregationResLevel
+  aggregationResLevel,
+  queryParameters
 }: Omit<FetchLayerDataParams, 'connection' | 'credentials'>) {
   const parameters = [encodeParameter('client', clientId || DEFAULT_CLIENT)];
 
   const sourceName = type === MAP_TYPES.QUERY ? 'q' : 'name';
   parameters.push(encodeParameter(sourceName, source));
 
+  if (queryParameters) {
+    parameters.push(encodeParameter('queryParameters', JSON.stringify(queryParameters || [])));
```

Comment:
`|| []` is not needed as it's already known to be truthy from condition in preceeding line